modules/xfem/src/userobjects/MeshCut3DUserObject.C

```diff
@@ -77,29 +77,37 @@ MeshCut3DUserObject::MeshCut3DUserObject(const InputParameters & parameters)
 void
 MeshCut3DUserObject::initialize()
 {
-  // this is a test of all methods relevant to front growth
+  _timestep = _fe_problem.timeStep();
+
   if (_grow)
   {
-    _stop = 0;
+    if (_timestep == 0)
+    {
+      findBoundaryNodes();
+      findBoundaryEdges();
+      sortBoundaryNodes();
+      refineBoundary();
+    }
```

Comment:
I'd prefer that we don't have these `_timestep` and `_last_timestep` member variables. It looks like the main (only?) reason for them is to do the special initialization stuff above once at the beginning of the analysis. You can probably do that by adding an `initialSetup()` method to this class, and putting it in there. `initialSetup()` only gets called once at the beginning of the analysis.